spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java

```diff
@@ -1066,6 +1068,50 @@ public void testDeleteWithMultipleSpecs() {
         sql("SELECT * FROM %s ORDER BY id", selectTarget()));
   }
 
+  @Test
+  public void testDeleteToWAPBranch() throws NoSuchTableException {
+    Assume.assumeTrue("WAP branch only works for table identifier without branch", branch == null);
```

Comment:
Do we fail when writing to a branch identifier if the `WAP_BRANCH` is set? If so, we could test that behavior here instead of using `Assume`.

Reply:
I see that this is tested in the next case, so we're good.